transports/websocket/src/framed.rs

```diff
@@ -540,6 +542,13 @@ fn parse_ws_dial_addr<T>(addr: Multiaddr) -> Result<WsAddress, Error<T>> {
         match protocols.pop() {
             p @ Some(Protocol::P2p(_)) => p2p = p,
             Some(Protocol::Ws(path)) => match protocols.pop() {
+                Some(Sni(domain)) => match protocols.pop() {
```

Comment:
If you don't import `Protocol::Sni`, this would be `Some(Protocol::Sni(domain))` which matches other uses of `Protocol` in this module.